merge-recursive.c

```diff
@@ -886,7 +887,8 @@ static int was_dirty(struct merge_options *o, const char *path)
 {
 	struct cache_entry *ce;
 
-	if (o->call_depth || !was_tracked(o, path))
+	if (o->call_depth || !was_tracked(o, path) ||
+	    is_excluded_from_virtualfilesystem(path, strlen(path), DT_REG) == 1)
```

Comment:
Thanks for this change, @dscho! This is really the thing we needed. And, I see that we have not send the virtualfilesystem code upstream, so we can't blame them for breaking it.